pkg/docker/adapter.go

```diff
@@ -257,32 +265,16 @@ func (d *DockerContainerAdapter) ApplyConfig(proposedConfig *agent.ProposedConfi
 		d.Logger().Warn("Restarting service")
 
 		// Execute docker restart command
-		err := d.dockerClient.ContainerRestart(context.Background(), d.ContainerName, container.StopOptions{})
+		err := d.dockerClient.ContainerRestart(ctx, d.Config.ContainerName, container.StopOptions{})
 		if err != nil {
 			return fmt.Errorf("failed to restart PostgreSQL service: %w", err)
 		}
 
 		// Wait for PostgreSQL to be back online with retries
-		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Minute)
-		defer cancel()
-
-		for {
-			select {
-			case <-ctx.Done():
-				return fmt.Errorf("timeout waiting for PostgreSQL to come back online")
-			case <-time.After(1 * time.Second):
-				// Try to execute a simple query
-				_, err := d.pgDriver.Exec(ctx, "SELECT 1")
-				if err == nil {
-					d.Logger().Info("PostgreSQL is back online")
-					return nil
-				}
-				d.Logger().Debug("PostgreSQL not ready yet, retrying...")
-			}
-		}
+		pg.WaitPostgresReady(d.PGDriver, ctx)
```

Comment:
Code re-use enabled, this block existed in at least 2 places, maybe more, I should double check this